posthog/hogql/database/database.py

```diff
@@ -286,7 +286,11 @@ def create_hogql_database(
     warehouse_tables: dict[str, Table] = {}
     views: dict[str, Table] = {}
 
-    for table in DataWarehouseTable.objects.filter(team_id=team.pk).exclude(deleted=True):
+    for table in (
+        DataWarehouseTable.objects.filter(team_id=team.pk)
+        .exclude(deleted=True)
+        .select_related("credential", "external_data_source")
```

Comment:
maybe this should really be 

```suggestion
        .select_related("credential", "external_data_source", "external_data_source__credentials")
```

Reply:
(although iirc you don't need to select_related the entire chain so...)
```suggestion
        .select_related("credential", "external_data_source__credentials")
```